src/main/scala/com/workday/warp/math/linalg/RidgeRegression.scala

```diff
@@ -0,0 +1,64 @@
+package com.workday.warp.math.linalg
+
+import org.apache.commons.math3.linear.{MatrixUtils, RealMatrix, SingularValueDecomposition}
+
+/**
+  * Analyzes multiple regression data that suffer from multicollinearity.
+  *
+  * When multicollinearity occurs, least squares estimates are unbiased, but their variances are large so they may be far
+  * from the true value. By adding a degree of bias to the regression estimates, ridge regression reduces the standard errors.
+  * It is hoped that the net effect will be to give estimates that are more reliable.
+  *
+  * Adapted from https://github.com/Netflix/Surus/blob/master/src/main/java/org/surus/math/RidgeRegression.java
+  *
+  * Created by tomas.mccandless on 9/26/18.
+  */
+class RidgeRegression(val x: Array[Array[Double]], val y: Array[Double], val l2Penalty: Double = 0.0001) {
+
+
+  val X: RealMatrix = MatrixUtils.createRealMatrix(this.x)
+  val X_svd: SingularValueDecomposition = new SingularValueDecomposition(this.X)
+
+  val (
+    coefficients: Array[Double],
+    fitted: Array[Double],
+    residuals: Array[Double],
+    standardErrors: Array[Double]
+  ) = this.updateCoefficients
+
+
+  /**
+    * Computes ridge regression.
+    *
+    * @return a tuple containing the result of ridge regression.
+    */
+  def updateCoefficients: (Array[Double], Array[Double], Array[Double], Array[Double]) = {
+    val V: RealMatrix = this.X_svd.getV
+    val s: Array[Double] = this.X_svd.getSingularValues.map(a => a / (a * a + l2Penalty))
+    val U: RealMatrix = this.X_svd.getU
+
+    val S: RealMatrix = MatrixUtils.createRealDiagonalMatrix(s)
+    val Z: RealMatrix = V.multiply(S).multiply(U.transpose)
```

Comment:
I know single letter references are more mathy and that this was adapted from other code, but it does make the code denser than it needs to be. At minimum, putting the math for this code up in the comment would help, or commenting inside the function with the gist of the operations (chunk them however you want).

Reply:
cool, i added some extra doc with the formula